IM_calculation/IM/intensity_measures.py

```diff
@@ -44,6 +46,36 @@ def get_spectral_acceleration_nd(acceleration, period, NT, DT):
         return get_spectral_acceleration(acceleration, period, NT, DT)
 
 
+def get_SDI(acceleration, period, DT, z, alpha, dy, dt):
+
+    return Bilinear_Newmark_withTH(
+        period,
+        z,
+        dy,
+        alpha,
+        acceleration * G / 100,  # Input is in m/s^2
+        DT,
+        dt,
+    ).T
+
+
+def get_SDI_nd(acceleration, period, NT, DT, z, alpha, dy, dt):
+    # SDI
+    if acceleration.ndim != 1:
```

Comment:
Have you tried running Bilinear Newmark with a 3 dimensional acceleration - it doesn't look like it's too hard to vectorise and it will significantly improve performance. (mind you this might be redundant for adv IMs; but thinking about it, probably will be directly applicable)

Reply:
I agree it is not difficult, but the performance is not much a big deal. At the moment, getting the correct result is more important (took a while for Vahid to confirm it works as intended) than the optimisation.